### ISSUE TITLE
Does flow_from_dataframe() support multilabel outputs?

### ISSUE THREAD

**Issue by scott-zockoll** [keras-team/keras-preprocessing#135]
I have previously been able to train a CNN to have multi-class and multi-label output using flow(), but I am having trouble getting it to train where the labels are vectors of binary values. I am not sure what class_mode I should be using? @Vijayabhaskar96 is this supported with flow_from_dataframe()?

**Comment by rragundez** [comment]
@scott-zockoll working on it today :)

**Comment by rragundez** [comment]
there you go https://github.com/keras-team/keras-preprocessing/pull/136
Hopefully it will get merge soon. A +1 is always appreciated in the PR.

**Comment by Vijayabhaskar96** [comment]
@scott-zockoll Sorry I was busy didn't notice the issue.
I have yet to see @rragundez 's PR You could change the multiple classes into one-hot vectors and split each column into separate columns such that the dataframe looks like this:
```
|filenames|class1|class2|class3|
|1.png|0|1|1|
|2.png|0|1|0|
|3.png|1|1|1|
...
```
You can perform this easily with Sklearn's [MultiLabelBinarizer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html)

You can now pass the `x_col="filenames" , y_col=["class1","class2","class3"] and class_mode="other"`
but this method only allows you to use a single Dense layer with 3 (no. of classes) neurons in the last layer with single loss.
If you want to use multiple loss functions you may want to write a python generator function that loops through the output of flow_from_dataframe and yields the reshaped array,this generator can be used now to fit the model with multiple losses.

**Comment by rragundez** [comment]
@Vijayabhaskar96 indeed I think that would work. Just something I wanted to point out:

By using the MultiLabelBinarizer or basically expanding the tools you impact the memory footprint of the algorithm without necessity, not only during the expansion process but also the class it self `DataFrameIterator` will have to keep in memory the full expanded matrix or dataframe of values for each observation and class. In many realistic scenarios this is actually quite a lot since many times you have tens or hundreds of classes and  thousand examples easily. In the PR this is optimize by keeping the minimum information necessary.

An option would be to support SparseDataFrame BTW, perhaps too far fetch.

**Comment by scott-zockoll** [comment]
@rragundez Great. I will check our your PR.

@Vijayabhaskar96 I see thank you. To fix the problem I had, I made some changes to support multi-label similar to this fork (but updated for the current keras implementation): https://github.com/tholor/keras/commit/29ceafca3c4792cb480829c5768510e4bdb489c5

**Comment by talhaanwarch** [comment]
what should be the class mode in case of multilabel?
